spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/HyperConformance.scala

```diff
@@ -93,11 +97,15 @@ object HyperConformance extends StreamTransformerFactory with HyperConformanceAt
   val log: Logger = LoggerFactory.getLogger(this.getClass)
 
   private val defaultReportVersion = 1
+  private val minimumSparkVersion: SemanticVersion = semver"2.4.3"
 
   @throws[IllegalArgumentException]
   override def apply(conf: Configuration): StreamTransformer = {
     log.info("Building HyperConformance")
 
+    SparkVersionGuard(minimumSparkVersion, SparkCompatibility.maxSparkVersionExcluded)
```

Comment:
Do we expect the Spark compatibility to differ between core DCE and HyperConformance?
If not, why not to use `fromDefaultSparkCompatibilitySettings`?

Reply:
Well the versions already differ. But we could have all versions requirements centralized somewhere. Thoughts?

Reply:
Ah, indeed. 2.4.2 vs. 2.4.3. Then let's have it centralized indeed. (`object SparkCompatibility`)

Reply:
So the minimum should be changed to 2.4.3?

Reply:
Just to make myself clearer. Centralized, as in one place, yes but we could keep multiple versions there. Like `minSparkVersionGeneral`, `minSparkVersionHyperConformance`, etc.

Question is, do we want to and how hard would it be to maintain?

Reply:
This would also mean adding a new entry to SparkVersionGuard

Reply:
I think more specifically, it would involve adding new field(s) to `SparkCompatibility`.

`SparkVersionGuard` case class and logic can stay the same, only calling it with different params should suffice. Either directly or via a different wrapper, e.g.
```scala
SparkVersionGuard.fromHyperConformaceSparkCompatibilitySetting
```